### DISCUSSION TITLE
[RFC] Open a new API for Table to implement custom Empty

### DISCUSSION THREAD

**Discussion by Wxh16144** [ant-design/ant-design#49603] [RFCs]
### 动机

现目前针对 Table 设置空状态有两种方式: 
1. `locale.emptyText`
2. `<ConfigProvider rednerEmpty={name=> name=== 'Table' ? Xxx : null} />`

方式 1 在文档中没找到明显说明地方。所以盲猜第二种方式也有不少人在用（我自己就是）

### 实施方案

新增一个 `empty`(命名待定)  属性。

1. 支持 `React.ReactNode` 允许开发者自定义显示空状态
2. 支持 `(source?: 'table' | 'table.filter')=> React.ReactNode`

### 详细说明

第一条类型没啥好说的，就是一个 React.ReactNode ， 也可以满足 https://github.com/ant-design/ant-design/pull/49599 的需求 <Table empty={null} />

第二条renderFunction 方式主要是为了解决 https://github.com/ant-design/ant-design/issues/49542 问题 ，具体解释说明移步：https://github.com/ant-design/ant-design/pull/49548#issuecomment-2190428255

### 缺点

没有，不影响之前使用

### 策略

Feature, minor version

**Comment by zombieJ** [comment]
做到 ConfigProvider 上，Table 本身不做额外支持。现在 Table empty 已经有了，再加就杂了。

**Comment by afc163** [comment]
Fixed in https://github.com/ant-design/ant-design/pull/49613
